x-pack/plugins/stack_connectors/server/connector_types/openai/openai.ts

```diff
@@ -51,6 +64,23 @@ export class OpenAIConnector extends SubActionConnector<Config, Secrets> {
     this.provider = this.config.apiProvider;
     this.key = this.secrets.apiKey;
 
+    this.openAI =
+      this.config.apiProvider === OpenAiProviderType.AzureAi
+        ? new OpenAI({
+            apiKey: this.secrets.apiKey,
+            baseURL: this.config.apiUrl,
+            defaultQuery: { 'api-version': getAzureApiVersionParameter(this.config.apiUrl) },
+            defaultHeaders: { 'api-key': this.secrets.apiKey },
+          })
+        : new OpenAI({
+            // default to empty string for unit tests x-pack/plugins/actions/server/integration_tests/connector_types.test.ts
+            apiKey: this.secrets.apiKey ?? '',
+            // without, this test fails x-pack/plugins/actions/server/integration_tests/connector_types.test.ts
+            // it seems the OpenAI sdk thinks the server emulation in jest is a browser environment
+            // without setting this flag, the sdk throws an error and the test will fail
+            dangerouslyAllowBrowser: true,
```

Comment:
@pmuellr this felt dirty. Please let me know if you have a better idea to address the problems brought forward by these snapshot tests: `x-pack/plugins/actions/server/integration_tests/connector_types.test.ts`

Reply:
Ya, hehehe.  I'd be willing to bet it's 100% safe, but also we'll be asked questions about until we eventually remove it.

I'm not super-happy with that test anyway, and am going to open an issue to improve it.  I'll mention this issue as well, there may be something we can do to convince Jest we're not browser-ish.

In the meantime, since these are only used directly, and not queued as alerting actions, let's just filter them out of the  tests in `x-pack/plugins/actions/server/integration_tests/connector_types.test.ts` - create a `Set<string>` of `connectorTypeId` values to **NOT** do the `getService()` test for.  The snapshot file will obviously change, but the diff should be easier to understand :-)

Reply:
I opened https://github.com/elastic/kibana/issues/179926 to track "fixing" the connector type tests ...

If you need some help filtering out the connectors from the existing test usage, LMK.